Us032 update readme (#43)
AmamMcMam/Team-A-Agile: README.md

# Team-A-Agile

## Table of Contents

> - Setup Instructions

Reviewer: ```suggestion
- How to run
    - Run backend
    - Run frontend
- Setup Instructions
```

    - Backend Setup
        - Java SDK
        - Intellij
    - Frontend Setup
        - Microsoft Visual Studio
- Testing Instructions
    - Unit Tests
    - Cucumber Tests

> 

Reviewer: Add commands/steps of how to run backend, frontend & what links to view them (http://localhost:8080/api and http://localhost:6555/)

## Setup Instructions

> ### Backend Setup

Reviewer: ```suggestion
```

> ####  **Java SDK**

Reviewer: ```suggestion
###  **Java SDK**
```

This project requires Java JDK version 16:
- [JDK 16 Download](https://www.oracle.com/java/technologies/javase-jdk16-downloads.html)

> ####  **Intellij**

Reviewer: ```suggestion
###  **Intellij**
```

For the backend you will need to install Intellij Community Edition if you don't already have it. Here is a link to the download page:

- [Intellij Download](https://www.jetbrains.com/idea/download/#section=mac)

You will then need to open the JavaWebService file in Intellij which is in:

 Team-A-Agile/backend

Once the project is opened the pom.xml file will run and install the necessary dependencies. If it doesn't then open this file in the editor and reload it by clicking the cycling arrow on the top right.

> 

Reviewer: ```suggestion
#### Backend set-up steps:
```

> You will need to create a file called "config.properties" in the resource folder to hold your database credentials. The resource folder is located in:

Reviewer: ```suggestion
1. You will need to create a file called "config.properties" in the resource folder to hold your database credentials. The resource folder is located in:
```

> src/main

Reviewer: ```suggestion
src/main/resources
```

Author: that path is referring to where the resources file is

Inside config.properties copy/paste the code below and replace the placeholder data with your own.

```
driver: placeholder
username: placeholder
password: placeholder
url: placeholder
```

> Next you will need to click on the drop-down menu in the top right next to the green hammer and click 'Edit Configurations'

Reviewer: ```suggestion
2. Next you will need to click on the drop-down menu in the top right next to the green hammer and click 'Edit Configurations'
```

![reference](images/setup1.png)

A menu will popup and in the 'program arguments' field enter in "server" click 'apply' then 'ok'.

You should be able to now run the backend by clicking the green arrow in the top right.

---

### Frontend Setup

####  **Microsoft Visual Studio**

> For the frontend you will need to install Microsoft Visual Studio Code.

Reviewer: You don't need VScode, maybe say that's what we recommend using

- [Microsoft Visual Studio Code download](https://code.visualstudio.com/?wt.mc_id=vscom_downloads)

> You will then need to open the 'frontend' file located in the root directory.

Reviewer: ```suggestion
1. Open the frontend folder
```

> In the 'frontend' folder create a file called ".env". Inside it copy/paste the code below replacing the 'placeholder' with the port number of the backend api. Make sure to save the file once added.

Reviewer: ```suggestion
2. In the root of the frontend folder create a file called ".env". 

+-- frontend
|   +-- .env

3. Inside the .env file, copy the code below replacing the 'placeholder' with the port number of the backend api. Make sure to save the file once added.
```

```
API_URL=http://localhost:placeholder/api
```

> Open the terminal in VScode by going to 'View' -> 'Terminal'

Reviewer: ```suggestion
4. Open terminal and type the following command:
```

Author: should add numbering but will keep the terminal instructions as not everyone will be familiar with the IDE

> ![reference](images/setup2.png)

Reviewer: not sure this image is needed

On the command line type "npm install"

Then once the packages have installed type "npm start".

> 

Reviewer: ```suggestion
```console
npm install
npm start
```

```

Author: I think its best to go step by step

> You now should be able to go to the localhost url, replace placeholder with your port number: http://localhost:placeholder/

Reviewer: ```suggestion
To view the frontend, follow this link: http://localhost:6555/
```
